src/irsForms/ScheduleEIC.ts

```diff
@@ -1,11 +1,349 @@
 
-export default class ScheduleEIC {
+type PrecludesEIC<F> = (f: F) => boolean
+
+const unimplemented = (message: string): void =>
+  console.warn(`[Schedule EIC] unimplemented ${message}`)
+
+const checks2555: PrecludesEIC<F2555> = (f): boolean => {
+  unimplemented('check F2555')
+  return false
+}
+
+const checks4797: PrecludesEIC<F4797> = (f): boolean => {
+  unimplemented('check F4797')
+  return false
+}
+
+const checks8814: PrecludesEIC<F8814> = (f): boolean => {
+  unimplemented('check F8814')
+  return false
+}
+
+const checksPub596: PrecludesEIC<Pub596Worksheet1> = (f): boolean => {
+  unimplemented('check Pub596 worksheet 1')
+  return false
+}
+
+const precludesEIC = <F>(p: PrecludesEIC<F>) => (f: F | undefined) => {
+  if (f === undefined) {
+    return false
+  }
+  return p(f)
+}
+
+export default class ScheduleEIC implements Form {
   tp: TaxPayer
+  f2555?: F2555
+  f4797?: F4797
+  f8814?: F8814
+  pub596Worksheet1: Pub596Worksheet1
+  qualifyingCutoffYear: number = 1996
+  investmentIncomeLimit: number = 3650
+
+  constructor (tp: TP) {
+    this.tp = new TaxPayer(tp)
+    this.f2555 = new F2555(tp)
+    this.f4797 = new F4797(tp)
+    this.f8814 = new F8814(tp)
+    this.pub596Worksheet1 = new Pub596Worksheet1(tp)
+  }
+
+  // At least key children, and max [NonMFJLimit, MFJLimit] income
+  step1ChildrenIncomes: Array<[number, number]> = [
+  // Nonmfj, mfjlimit
+    [15820, 21710], // 0 dependents
+    [41756, 47646], // 1
+    [47440, 53330], // 2
+    [50954, 56844] // 3
+  ]
+
+  // instructions step 1.1
+  passIncomeLimit = (f1040: F1040): boolean => {
+    if (this.tp.tp.filingStatus !== undefined) {
+      const incomeLimits = federal.EIC.caps[this.tp.tp.filingStatus]
+      if (incomeLimits !== undefined) {
+        const limit = incomeLimits[Math.min(this.qualifyingDependents.length, incomeLimits.length - 1)]
+        return (f1040.l11() ?? 0) < limit
+      }
+    }
+    return false
+  }
+
+  // Step 1.2, todo, boths spouses must have a SSN issued before 2020 due date
+  // and without work restriction and valid for eic purposes
+  validSSNs = (): boolean => {
+    unimplemented('Step 1.2 (valid SSNs) unchecked')
+    return true
+  }
+
+  // Step 1.3
+  allowedFilingStatus = (): boolean => this.tp.tp.filingStatus !== FilingStatus.MFS
+
+  // Step 1.4
+  allowedFilling2555 = (): boolean => !precludesEIC(checks2555)(this.f2555)
+
+  // Step 1.5 nonResidentAlien
+  allowedNonresidentAlien = (): boolean => {
+    unimplemented('Step 1.5, Not checking non-resident alien')
+    return true
+  }
+
+  // step 2, question 1
+  investmentIncome = (f1040: F1040): number => sumFields([
+    f1040.l2a(),
+    f1040.l2b(),
+    f1040.l3b(),
+    Math.max(f1040.l7() ?? 0, 0)
+  ])
+
+  passInvestmentIncomeLimit = (f1040: F1040): boolean =>
+    this.investmentIncome(f1040) < federal.EIC.maxInvestmentIncome
+
+  // Todo, step 2, question 3
+  f4797AllowsEIC = (): boolean => !precludesEIC(checks4797)(this.f4797)
+
```

Comment:
@thegrims Here are some todo items, for now they are console warnings if the calculations are attempted.